packages/skill-api/src/core/services/redeem-golden-ticket.ts

```diff
@@ -48,14 +48,16 @@ export async function redeemGoldenTicket({
     // something in the chain strips out the plus and leaves a space
     const email = String(baseEmail).replace(' ', '+')
 
-    const coupon = await getCoupon({
-      where: {id: couponId},
-    })
+    const coupon = await getCouponWithBulkPurchases(couponId)
 
     const couponValidation = validateCoupon(coupon)
 
     if (coupon && couponValidation.isRedeemable) {
-      const bulkCouponRedemption = coupon.maxUses > 1
+      // if the Coupon is the Bulk Coupon of a Bulk Purchase,
+      // then a bulk coupon is being redeemed
+      const bulkCouponRedemption = Boolean(
+        coupon?.bulkCouponPurchases[0].bulkCouponId,
```

Comment:
This probably needs to be more like:

```
coupon?.bulkCouponPurchases?.[0]?.bulkCouponId
```

Reply:
`first` from lodash is generally my goto and it handles all that under the hood